packages/osd-std/src/index.ts

```diff
@@ -29,13 +29,15 @@
  */
 
 export { assertNever } from './assert_never';
-export { deepFreeze, Freezable } from './deep_freeze';
+export { deepFreeze } from './deep_freeze';
+export type { Freezable } from './deep_freeze';
```

Comment:
How will this distinction between type exports be enforced/detected in future packages?

Reply:
Without this, the built artifact for `index.ts` would have
```js
// packages/osd-std/target/node/index.js
Object.defineProperty(exports, "Freezable", {
  enumerable: true,
  get: function () {
    return _deep_freeze.Freezable;
  }
});
```
... for the CommonJS builds but `_deep_freeze` doesn't export `Freezable` in their JS, or
```
// packages/osd-std/target/web/index.js
export { deepFreeze, Freezable } from './deep_freeze';
```
... for ES modules but `./deep_freeze` doesn't export `Freezable`.

By telling the compilers that this is a `type` export, we get them to not re-export it. Of course, no one uses them in the compiled code but they are misleading and take space.

Reply:
Oh yeah, that part I get - what I was asking is whether there's anything preventing a future dev unfamiliar with that distinction from making a similar mistake and exporting/re-exporting the type needlessly. But it sounds like the answer is that it's not really a problem, so we're just fixing the known instances.

Reply:
Since I am slightly obsessed with optimization, I would call it a problem but I am sure I would be in the minority if I wanted to prioritize this. I am biting my tongue for when the typescript upgrade happens; just choosing my battles.

Reply:
OK, that's fine. It's not like we're frequently adding or creating new packages to the project, was just trying to avoid the problem coming back.